sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -483,8 +484,16 @@ trait CheckAnalysis extends PredicateHelper {
                   checkColumnNotExists("add", add.fieldNames(), table.schema)
                 }
                 val parent = findParentStruct("add", add.fieldNames())
-                positionArgumentExists(add.position(), parent)
+                add.position match {
+                  case after: After =>
+                    // Handle the case where column position is referencing new columns being added.
+                    if (!colsToAdd.contains(add.fieldNames().init :+ after.column)) {
+                      positionArgumentExists(add.position(), parent)
```

Comment:
shall we just add one more parameter to `positionArgumentExists` for the newly added columns?

Reply:
just want to make sure all the caller side of `positionArgumentExists` take care of newly added columns.